rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,97 @@
+are selected and the difference in the expression level for each gene is computed by comparing the average
+expression level for each set of cells. When there is a particularly large difference in expression between the two sets of
+cells, that gene is identified and presented to the user. To ensure consistency and compatibility it is necessary to define a
+data format for storing references to these genes and any accompanying statistical data.
+
+Product Requirements
+
+### Users should be able to export differential expression results
+
+Currently cellxgene users can choose two sets of cells and compute/display the log fold and adjusted p-value of the top ten differentially expressed genes. Additionally we display a histogram of the gene expression levels of the selected cells for each gene in the right side bar.
+
+- [US1] A user should be able to export a gene set, if the gene set was created by differential expression, the pvalue, log fold change, and category label fields will be populated in addition to the gene name. This will be available in a csv file (see data formats below for more detail).
+
+### Users should be able to upload their own gene sets and see them in the right side bar with the option to apply them within current cellxgene functionalities, if the same gene ontology was used when generating the current dataset and the geneset uploaded by the user (ie the gene names match)
+
+- [US2] A user should be able to upload a geneset via a csv file (see data formats below for more detail) for a dataset.
+- [US3] A user should be able to upload a geneset they previously downloaded from cellxgene
+- [US4] When a user uploads a geneset, that list of genes should appear in the right side bar
+- [US5] If the user was logged in when they uploaded a geneset that genesit should persist if they logout and back in or reload the page
+- [US6] If the gene names match those used in the dataset the user should be able to use general cellxgene functionality (color by genes, calculate differential expression levels, etc.) on those genes
+
+## Detailed Design | Architecture | Implementation
+
+### Data model
+
+Fully supporting persistence of differential expression results is out of the scope of this RFC. But the data model described below has been designed to allow for differential expression persistence in the future.
+
+### Relational Database
+
+To support gene sets three tables will be added to the cellxgene relational database
+
+- Gene
+  - UUID (generated)
+  - GeneName
+  - OntologyId?
+- GeneGeneSetLink
+  - UUID
+  - Gene UUID
+  - GeneSet UUID
+  - Comments (optional)
+- GeneSet
+  - UUID (generated)
+  - GeneSet Name (required)
+  - Gene Count (generated)
+  - User UUID
+  - Dataset UUID
+
+![Cellxgene Data Schema](imgs/Cellxgene_rds_schema.png)
+**Figure 1** Cellxgene data schema, tables to be added (as described above) are in red
+
+#### Imported CSV
+
+GeneSet CSV
```

Comment:
It might be worth stating it clearly that a CSV file can contain multiple gene sets.  I wasn't clear on that at first.
When we export a CSV file will we output files with multiple gene sets?  
I guess that probably depends on how the user interface is designed.

Reply:
Yes, the input and output CSV will both contain arbitrarily many genesets, containing arbitrarily many genes

Reply:
@colinmegill I've been assuming the exported csv will typically contain one geneset (whatever the user is currently working with). Are you thinking we will export multiple genesets? How will we determine which ones?

Reply:
If the user can point to a geneset and click 'copy member genes' in the cellxgene interface, they are good to go. If they are doing this from the data portal side this seems like a less likely pattern.

Reply:
Download all genesets to CSV sounds like an option at the top of the right sidebar. Then individual genes could have copy members with header.